pandas/_libs/ops.pyx

```diff
@@ -292,15 +292,19 @@ def maybe_convert_bool(ndarray[object] arr,
         elif val in false_vals:
             result[i] = 0
         elif isinstance(val, float):
-            result[i] = UINT8_MAX
-            na_count += 1
+            mask[i] = 1
```

Comment:
what if it is a float other than NaN?

Reply:
Nice catch, this also seems buggy in the original code. The solution would probably be an is_nan check and returning arr otherwise.